pkgs/applications/window-managers/sway/beta.nix

```diff
@@ -0,0 +1,43 @@
+{ stdenv, fetchFromGitHub
+, meson, ninja
+, pkgconfig, asciidoc, libxslt, docbook_xsl
+, wayland, wlc, libxkbcommon, pcre, json_c, dbus
+, pango, cairo, libinput, libcap, pam, gdk_pixbuf, libpthreadstubs
+, libXdmcp, wlroots, wayland-protocols
+, buildDocs ? true
+}:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "sway-beta";
+  version = "1.0-beta.1";
+
+  src = fetchFromGitHub {
+    owner = "swaywm";
+    repo = "sway";
+    rev = version;
+    sha256 = "0h9kgrg9mh2acks63z72bw3lwff32pf2nb4i7i5xhd9i6l4gfnqa";
+  };
+
+  nativeBuildInputs = [
+    pkgconfig meson ninja
+  ] ++ stdenv.lib.optional buildDocs [ asciidoc libxslt docbook_xsl ];
+
+  buildInputs = [
+    wayland wlc libxkbcommon pcre json_c dbus
+    pango cairo libinput libcap pam gdk_pixbuf libpthreadstubs
+    libXdmcp wlroots wayland-protocols
+  ];
```

Comment:
You can drop `wlc`. I'm not sure if `libXdmcp` and `libpthreadstubs` are still required (it builds fine without them but I didn't verify if they're really not required - they aren't listed [here](https://github.com/swaywm/sway#compiling-from-source) as well).